src/clinicalMongo/index.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  MONGO_PASS,
+  MONGO_URL,
+  MONGO_USER,
+  USE_VAULT,
+  VAULT_MONGO_SECRET_PATH
+} from "config";
+import mongoose from "mongoose";
+import { loadVaultSecret, createVaultClient } from "vault";
+import logger from "logger";
+
+type MongoSecret = {
+  content: string;
+};
+
+type MongoSecretContent = {
+  CLINICAL_DB_USERNAME: string;
+  CLINICAL_DB_PASSWORD: string;
+};
+
+const isMongoSecret = (obj: { [k: string]: any }): obj is MongoSecret => {
+  return typeof obj["content"] === "string";
+};
+
+const isMongoSecretContent = (obj: {
+  [k: string]: any;
+}): obj is MongoSecretContent => {
+  return (
+    typeof obj["CLINICAL_DB_USERNAME"] === "string" &&
+    typeof obj["CLINICAL_DB_PASSWORD"] === "string"
+  );
+};
+
+export default async ({
+  vaultClient = createVaultClient(),
```

Comment:
I'm just trying to understand will this run as soon as you call an imported instance and what happens if it fails

Reply:
oh actually I guess it will be called when you try to use the vaultClient, which will barf if not configured correctly...

Reply:
`createVaultClient` will get called if you call this function without passing in a `vaultClient`. Otherwise, it won't be